llvm/lib/Transforms/IPO/LowerTypeTests.cpp

```diff
@@ -1960,8 +1967,9 @@ static void dropTypeTests(Module &M, Function &TypeTestFunc) {
     // phi (which will feed the assume). Simply replace the use on the phi
     // with "true" and leave the merged assume.
     if (!CI->use_empty()) {
-      assert(
-          all_of(CI->users(), [](User *U) -> bool { return isa<PHINode>(U); }));
+      assert(AlwaysDrop || all_of(CI->users(), [](User *U) -> bool {
```

Comment:
Happy to drop the assert instead of passing through all the extra booleans, but I wasn't sure if that was somehow load bearing outside of the FatLTO use case.